bioblend/_tests/TestGalaxyWorkflows.py

```diff
@@ -205,15 +205,64 @@ def test_run_workflow(self):
             self.gi.workflows.run_workflow(wf['id'], None)
 
     def test_invoke_workflow(self):
+        invoke_response = self._invoke_workflow()
+        assert invoke_response['state'] == 'new', invoke_response
+
+    def test_create_workflow_from_history(self):
+        invoke_response = self._invoke_workflow()
+        wf_id = invoke_response['workflow_id']
+        history_id = invoke_response['history_id']
+        self.gi.workflows.cancel_invocation(wf_id, invoke_response['id'])
+        wf2 = self.gi.workflows.create_workflow_from_history(history_id, 'My new workflow!')
+        self.assertEqual(wf2['name'], 'My new workflow!')
```

Comment:
Can you make the test a bit tighter here? E.g. by comparing the number of steps to the original workflow.

Btw, I think you need to submit the `job_ids` and other parameters, otherwise you get an empty workflow.

Reply:
Sorry, I cannot get it to work. Passing job_ids gives me "Attempt to create workflow with job not connected to current history". Passing dataset_ids results in a workflow with no steps.

Reply:
Can you try something like `job_ids = [job['id'] for job in self.gi.jobs.get_jobs() if job['history_id'] == history_id]`?

For the datasets, you should pass the hid, not the id, i.e. the index of the dataset in the history. So `[1, 2]` should be enough.

Reply:
I tried it but it makes no difference to the output.

Reply:
After a bit of fiddling, it's once again a time issue - the invocation needs to fully complete before the job_ids can be collected (see the latest commit).The `Attempt to create workflow with job not connected to current history` is misleading.

Can you ensure _wait_invocation waits until all individual steps are scheduled, not just the invocation itself? That should hopefully fix it.

Reply:
Another point: we should not duplicate the _wait_invocation method. Can you move this to GalaxyTestBase.py to reduce boilerplate?

Reply:
> After a bit of fiddling, it's once again a time issue - the invocation needs to fully complete before the job_ids can be collected (see the latest commit).The `Attempt to create workflow with job not connected to current history` is misleading.
> 
> Can you ensure _wait_invocation waits until all individual steps are scheduled, not just the invocation itself? That should hopefully fix it.

@simonbray From my reading of Galaxy's `lib/galaxy/workflow/run.py` an invocation state should be set to `scheduled` only when all the steps have also been scheduled (though the jobs may still have to be).

Reply:
I removed the check of steps in `wait_for_invocation`.